### PR TITLE
fix: update to use the old JSX transform

### PR DESCRIPTION
**Pull request by zchenwei**
#### Description of changes

Update to use the old JSX transform to fix a regression.

#### Description of how you validated changes

Create a sample React app running on a React version prior to 16.14.0 and integrate ui-react library into the app

| Before | After |
| ----------- | ----------- |
| Screen Shot 2022-08-25 at 4 04 06 PM | Screen Shot 2022-08-25 at 4 09 15 PM |

#### Checklist

- [x] PR description included
- [x] `yarn test` passes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react/scripts/generateIcons.js [hunk 1]
```diff
@@ -8,6 +8,7 @@ const iconNames = [];
 
 const template = ({ iconName, source }) => {
   return `import classNames from 'classnames';
+  import * as React from 'react';
```
Comment (zchenwei) [inline packages/react/scripts/generateIcons.js:11]: The old transform will need React in scope so that why I added it in a few places

**Comment by calebpollman** [comment]
Thanks for handling this! Looks straightforward to me, but can you cherry pick the commit and test against [in-app-messaging/main](https://github.com/aws-amplify/amplify-ui/tree/in-app-messaging/main) please?

File: .changeset/rich-peas-try.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@aws-amplify/ui-react": patch
+---
+
+fix: update to use the old JSX transform to support app running on React prior to 16.14.0 where the nex JSX transform does not get backported
```
Comment (joebuono) [inline .changeset/rich-peas-try.md:5]: nit typo: "next JSX"

**Review by joebuono — DISMISSED** [review]
LGTM! Thanks for fixing this

File: .changeset/rich-peas-try.md (hunk 2 above)
Comment (zchenwei) [reply .changeset/rich-peas-try.md:5]: ohhh, good catch I mean `new`

**Comment by zchenwei** [comment]
> Thanks for handling this! Looks straightforward to me, but can you cherry pick the commit and test against [in-app-messaging/main](https://github.com/aws-amplify/amplify-ui/tree/in-app-messaging/main) please?

Yeah man, I test against that branch you mention in a sample app and it looks good. For the build, since the old transform need React in scope, so anywhere you use JSX you will need to import React in the module and Rollup will hint you. There is only one place I can see you need to update during my testing:
`packages/react/src/components/InAppMessaging/withInAppMessaging/__tests__/withInAppMessaging.spec.tsx`

**Review by joebuono — APPROVED** [review]

File: packages/react/tsconfig.json [hunk 3]
```diff
@@ -4,7 +4,7 @@
     "baseUrl": ".",
     "declaration": true,
     "esModuleInterop": true,
-    "jsx": "react-jsx",
```
Comment (wlee221) [inline packages/react/tsconfig.json:7]: For my knowledge, can you briefly explain why this had to be changed?

Comment (zchenwei) [reply packages/react/tsconfig.json:7]: This is controls how JSX constructs are emitted in JavaScript files. It has better details here <- https://www.typescriptlang.org/tsconfig#jsx

Comment (wlee221) [reply packages/react/tsconfig.json:7]: TIL!

**Review by wlee221 — APPROVED** [review]

**Review by calebpollman — APPROVED** [review]
LGTM 🚢

**Review by reesscot — APPROVED** [review]

**Pull request merged**